vshard.go

```diff
@@ -122,8 +124,34 @@ type Config struct {
 }
 
 type BucketStatInfo struct {
-	BucketID uint64 `mapstructure:"id"`
-	Status   string `mapstructure:"status"`
+	BucketID uint64 `msgpack:"id"`
+	Status   string `msgpack:"status"`
+}
+
+// tnt vshard storage returns map with 'int' keys for bucketStatInfo,
+// example: map[id:48 status:active 1:48 2:active].
+// But msgpackv5 supports only string keys when decoding maps into structs,
+// see issue: https://github.com/vmihailenco/msgpack/issues/372
+// To workaround this we decode BucketStatInfo manually (using mapstructure).
+// When the issue above will be resolved, this code can be (and should be) deleted.
+func (bsi *BucketStatInfo) DecodeMsgpack(d *msgpack.Decoder) error {
+	v, err := d.DecodeInterface()
```

Comment:
using Mapstructure in any form will be less profitable than native parsing

Reply:
Fixed